### PR TITLE
SharePoint 2013 artifacts

### PR DESCRIPTION
**Pull request by RogerBestMsft**
Adding SharePoint 2013 installation artifacts (4). 
- Install prerequisites and handle reboot (A & B).
- Setup SharePoint 2013 (C).
- Configure Sharepoint 2013 with a Dev site (D).

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@RogerBestMsft__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (rbest). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: Artifacts/windows-sharepoint2013A/Artifactfile.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+  "$schema": "https://raw.githubusercontent.com/Azure/azure-devtestlab/master/schemas/2015-01-01/dtlArtifacts.json",
```
Comment (drwill-ms) [inline Artifacts/windows-sharepoint2013A/Artifactfile.json]: Please update to the latest schema version: 2016-11-28

File: Artifacts/windows-sharepoint2013A/Artifactfile.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+  "$schema": "https://raw.githubusercontent.com/Azure/azure-devtestlab/master/schemas/2015-01-01/dtlArtifacts.json",
+  "title": "SharePoint Server 2013 (1) Prerequisites installer",
```
Comment (drwill-ms) [inline Artifacts/windows-sharepoint2013A/Artifactfile.json]: What does the (1) mean?

File: Artifacts/windows-sharepoint2013A/Artifactfile.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  "parameters": {
+    "downloadUrl": {
+      "type": "string",
+      "displayName": "Download URL for SharePoint",
+      "description": "The download URL for SharePoint 2013.",
+      "defaultValue": "https://download.microsoft.com/download/6/E/3/6E3A0B03-F782-4493-950B-B106A1854DE1/sharepoint.exe"
+    }
+  },
+  "runCommand": {
+    "commandToExecute": "[concat('powershell.exe -ExecutionPolicy bypass -File SharePoint2013SP1P1.ps1', ' -downloadUrl ', parameters('downloadUrl'))]"
```
Comment (drwill-ms) [inline Artifacts/windows-sharepoint2013A/Artifactfile.json]: No need to have ' -downloadUrl ' as a separate parameter from the previous one.

Comment (drwill-ms) [inline Artifacts/windows-sharepoint2013A/Artifactfile.json]: We have found that using the -File parameter prevents errors flagging the artifact as failed to install. Instead use:

powershell.exe -ExecutionPolicy bypass \"& .\yourscript.ps1 -parameters\"

File: Artifacts/windows-sharepoint2013A/Artifactfile.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    "Windows"
+  ],
+  "iconUri": "https://i-msdn.sec.s-msft.com/dynimg/IC792065.jpg",
+  "targetOsType": "Windows",
+  "parameters": {
+    "downloadUrl": {
+      "type": "string",
+      "displayName": "Download URL for SharePoint",
+      "description": "The download URL for SharePoint 2013.",
+      "defaultValue": "https://download.microsoft.com/download/6/E/3/6E3A0B03-F782-4493-950B-B106A1854DE1/sharepoint.exe"
```
Comment (drwill-ms) [inline Artifacts/windows-sharepoint2013A/Artifactfile.json]: How likely is this URL to change in the future?

File: Artifacts/windows-sharepoint2013B/Artifactfile.json [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+  "$schema": "https://raw.githubusercontent.com/Azure/azure-devtestlab/master/schemas/2015-01-01/dtlArtifacts.json",
+  "title": "SharePoint Server 2013 (2) Prerequisites installer",
+  "description": "Installs SharePoint Server 2013 SP1 Prerequisites part 2.",
+  "tags": [
+    "Windows"
+  ],
+  "iconUri": "https://i-msdn.sec.s-msft.com/dynimg/IC792065.jpg",
+  "targetOsType": "Windows",
+  "runCommand": {
+    "commandToExecute": "powershell.exe -ExecutionPolicy bypass -File SharePoint2013SP1P2.ps1"
```
Comment (drwill-ms) [inline Artifacts/windows-sharepoint2013B/Artifactfile.json]: Same as above, don't use -File

File: Artifacts/windows-sharepoint2013B/SharePoint2013SP1P2.ps1 [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+$localFile = Join-Path $path $preInstaller
+$localLog = Join-Path $path $logFile
+$localreboot = Join-Path $path $rebootTemp
+
+If (Test-Path $localreboot){
+
+    Remove-Item -Path $localreboot
+
+    Add-content $localLog -value "Starting PreRequisite /continue process"
```
Comment (drwill-ms) [inline Artifacts/windows-sharepoint2013B/SharePoint2013SP1P2.ps1]: No capital R in prerequisite

File: Artifacts/windows-sharepoint2013B/SharePoint2013SP1P2.ps1 [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+    #Extract sharePoint Files from downloaded file
+
+    $argumentList = " /unattended /continue"
+
+    $retCode = Start-Process -FilePath $localFile -ArgumentList $argumentList -Wait -PassThru
+
+    if ($retCode.ExitCode -eq 0)
+    {
+        Add-content $localLog -value "SharePoint 2013 prerequisite installer 2 succeded - no reboot"
```
Comment (drwill-ms) [inline Artifacts/windows-sharepoint2013B/SharePoint2013SP1P2.ps1]: If no reboot is required, by unconditionally make the call to reboot on line 52?

File: Artifacts/windows-sharepoint2013C/Artifactfile.json [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+    "parameters": {
+      "PIDKEY": {
+        "type": "string",
+        "displayName": "Product Key",
+        "description": "*Not required* Leave the Product key as zero to install the trial version.",
+        "defaultValue" :  "0"
+      }
+  },
+  "runCommand": {
+	"commandToExecute": "[concat('powershell.exe -ExecutionPolicy bypass -File SharePoint2013SP1Setup.ps1', ' -pidKey ', parameters('PIDKEY'))]"
```
Comment (drwill-ms) [inline Artifacts/windows-sharepoint2013C/Artifactfile.json]: Same as above, don't use -File.

File: Artifacts/windows-sharepoint2013C/ChangeNetVersion.ps1 [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+$key1033Release = $key1033.GetValue("Release")
+$key1033Version = $key1033.GetValue("Version")
+
+#if the incorrect version change
+
+if ($keyRelease -ne '378389')
+{
+    $key.SetValue('Release','378389',[Microsoft.Win32.RegistryValueKind]::DWord)
+    $key1033.SetValue('Release','378389',[Microsoft.Win32.RegistryValueKind]::DWord)
+    Add-content 'C:\junker.txt' -value $keyRelease
```
Comment (drwill-ms) [inline Artifacts/windows-sharepoint2013C/ChangeNetVersion.ps1]: What is junker.txt? Why is it in the root of c:?

File: Artifacts/windows-sharepoint2013C/RevertNetVersion.ps1 [hunk 10]
```diff
@@ -0,0 +1,21 @@
+﻿
+$keyfile = 'C:\junker.txt'
```
Comment (drwill-ms) [inline Artifacts/windows-sharepoint2013C/RevertNetVersion.ps1]: Maybe in the temp directory?

**Pull request closed without merging**